games-emulation/bsnes_libretro/patches/bsnes_libretro-0.94.git20171227.patchset

```diff
@@ -1,14 +1,14 @@
-From 56a7340a68473a078cc321d4c50bf8a0d1628efd Mon Sep 17 00:00:00 2001
-From: Puck Meerburg <puck@puckipedia.nl>
-Date: Sat, 17 Jan 2015 19:54:22 +0100
-Subject: Fix Haiku build
+From 4236f2967ce88e391cc985f78bf81158032cf20e Mon Sep 17 00:00:00 2001
+From: kwyxz <kwyxz@kwyxz.org>
```

Comment:
It would be nice to preserve the authorship of the part of the patchset that already existed (even if you update the line offsets), and add a new commit in the patchset for the changes you added.
This can be done with:
```
haikuporters -B bsnes_libretro-0.94.git20171227
cd work-0.94*/sources/*
```
`haikuporters -B` applies the existing patchset, and does its best to workaround any offset mismatches.
Then you apply your changes to `nall/platform.hpp` and `target-libretro/Makefile` and continue with:
```
git commit nall/platform.hpp target-libretro/Makefile
cd ../../..
cd patches
cp bsnes_libretro-0.94.git20171227.patchset bsnes_libretro-0.94.git20171227.patchset.mybackup
haikuporters -e bsnes_libretro-0.94.git20171227
```
`haikuporters -e` will sync the patch offsets.

Another solution would be to manually edit the patchset (in order to insert a git header before your changes), then clean the work directory, apply it with `haikuporter -B` and regenerate it with `haikuporter -e` (to fix any minor formating issues.)

Reply:
Ha I did not know this was possible. Let me see what I can do to (rightfully so) preserve the authorship.

Reply:
Hmmm `hp -B` is not going so well

````
Applying patchset "/boot/home/haikuports/games-emulation/bsnes_libretro/patches/bsnes_libretro-0.94.git20171227.patchset" ...
error: sha1 information is lacking or useless (Makefile).
error: could not build fake ancestor
Command '['git', 'am', '--ignore-whitespace', '-3', '--keep-cr', u'/boot/home/haikuports/games-emulation/bsnes_libretro/patches/bsnes_libretro-0.94.git20171227.patchset']' returned non-zero exit status 128
````

I guess I might need to manually edit these files.

Reply:
In that case you can comment out the `PATCHES=` in the recipe, copy to a safe place each file that gets patched, and then:
```
haikuporters -c bsnes_libretro-0.94.git20171227
haikuporters -B bsnes_libretro-0.94.git20171227
```
overwrite the files that were being patched by the old patch, do a git commit on them, then overwrite the files that you were not yet being patched by the old patch, do a new commit and:
```
haikuporters -e bsnes_libretro-0.94.git20171227
```
After that, you will just need to replace the very first 4 lines of the generated patchset by the very first 4 lines of the old patchset.
Once your patchset is ready, you should uncomment the `PATCHES=`.
BTW, after this, in case you plan to continue patching, you shoud run
```
haikuporters -c bsnes_libretro-0.94.git20171227
haikuporters -B bsnes_libretro-0.94.git20171227
```
again, so that you won't need to manually edit the patchset everytime you regenerate the patchset with
```
haikuporters -e bsnes_libretro-0.94.git20171227
```

Reply:
To summerize my previous comment, the method consist in:
1) copying to a safe place each of the files you have patched so far.
2) clean the work directory with `hp -c ...`
3) change the recipe so that hp thinks it does not have any patch.
4) go to the build directory
5) overwrite each file (that needs to be patched) with the copies you did at step 1, and commit the changes inside the build directory.
6) leave the build directory (I don't know if this is required, but it does not hurt :)
7) extract the patchset with `hp -e ...`
8) uncomment the `PATCHES=` in the recipe.

You shoud repeat step 6 as many times as you would like to have have git headers in your patchset. Of course you can commit more than one patch at a time.
You can even "host" changes to the same file in different commits. That can be handy if a previous author has made changes in e.g. some part of a file and you need to add more changes to another part of that same file.

Reply:
How does it look like now? Builds fine for me.